### ISSUE TITLE
Questions about submitting kitti official website

### ISSUE THREAD

**Issue by Senwang98** [SPengLiang/DID-M3D#10]
I'm working on my master's paper of Mono3d. 
I'd like to know if you use all the `trainval` datasets when you submit to the kitti official website? 
Then how do you determine which ckpt is the best one?

**Comment by SPengLiang** [comment]
You are right. When evaluating on KITTI website, you should use the trainval set, and you can use the final ckpt or build a small validation set (e.g., 100 samples) for selecting the best one.  Good Luck.

**Comment by Senwang98** [comment]
@SPengLiang 
Ok, thanks.

**Comment by Senwang98** [comment]
@SPengLiang 
Sorry to disturb you again.
I want to know how 16.29mAP is trained?
In your paper, it seems that you only report `car` category performance in KITTI-test.

Because it's my first time to submit to kitti official, I guess you only trained the `Car` when using trainval dataset, so that the test output only include car class.
I think use only car category to train, the final performance maybe higher, if I want to train only `Car` category, I only need to modify the `self.num_classes=1` and `self.writelist=['Car']`?
Do you use `class_merging` and `use_dontcare` to train final model?

My results:
![image](https://user-images.githubusercontent.com/23122139/196861292-1444f3b1-bce8-48b8-a4fc-2c197986dba9.png)

**Comment by SPengLiang** [comment]
You are right, we only trained the Car when using trainval dataset (we do not use "class_merging" and "use_dontcare" to train final model). In our experiments, this manner obtains better performance on Car on KITTI test set. Additionally, we submitted our results two times on KITTI test set, reporting the better one. We find that results on KITTI test set are not very stable, even though we trained the same model twice. Please note that there is a submission limitation on KITTI (3 times in 30 days). Please feel free to contact me if you have further questions.

**Comment by Senwang98** [comment]
@SPengLiang 
Thanks for your reply!
I modify the following code so that I can train only car class, can you help me to check it, thanks very much!!
![image](https://user-images.githubusercontent.com/23122139/196864709-0c9780bc-c86b-4019-91fe-1794e6b4d551.png)
I only modify these three lines of code to meet my need.

**Comment by SPengLiang** [comment]
I have the impression that you only need to modify the "writelist" in "config/kitti.yaml", from "['Car','Pedestrian','Cyclist']" to "['Car']", others keep the same to original codes.

<img width="508" alt="image" src="https://user-images.githubusercontent.com/14855845/196865252-fbc8cef8-b922-4846-95ff-4bf4f447c979.png">

**Comment by Senwang98** [comment]
@SPengLiang 
Got it, I will try it. 
you are welcome and thank you again!!!
